src/components/secret-display/SecretDisplaySection.js

```diff
@@ -92,12 +93,15 @@ export default function SecretDisplaySection({
   const walletId = params?.walletId || selectedWallet.id;
   const currentWallet = wallets[walletId];
   const [visible, setVisible] = useState(true);
+  const [isHideRecoveryPhrase, setHideRecoveryPhrase] = useState(false);
   const [seed, setSeed] = useState(null);
   const [type, setType] = useState(currentWallet?.type);
 
   const loadSeed = useCallback(async () => {
     try {
-      const s = await loadSeedPhraseAndMigrateIfNeeded(walletId);
+      const { seed: s, actionStatus } = await loadSeedPhraseAndMigrateIfNeeded(
```

Comment:
this flow is a bit funky. I see what you're doing, but it's hard to follow.
Maybe this is a good moment to just throw an error inside `src/handlers/authentication.ts` and handle it here? Cause this is actually error what happened 

Or maybe something else, just I have a preference for not "replicating" the structure of add data struct just for reducing type-handling